### PR TITLE
Allow the pool lock to be specialized

### PR DESCRIPTION
**Pull request by harlowja**
It can be quite useful to allow the lock that the pool
uses to be specified to be something other than the default
threading lock class/type. This can be especially useful when
eventlet (or gevent or other) are used which have there own
special lock type that connects into there event-loop.

### CODE REVIEW

File: pymemcache/client.py [hunk 1]
```diff
@@ -836,7 +836,10 @@ def __init__(self,
                  ignore_exc=False,
                  socket_module=socket,
                  key_prefix=b'',
-                 max_pool_size=None):
+                 max_pool_size=None,
+                 # Allow this to be a different style of lock (eventlet
+                 # lock/semaphore for example, that works with greenthreads).
```
Comment (jparise) [inline pymemcache/client.py]: This type of documentation generally goes in a docblock.

Comment (harlowja) [inline pymemcache/client.py]: Ya. I'll add those into a docstring/block on the class.

**Comment by cgordon** [comment]
This is pretty nit-picky, but we've been using the Google style guide for documentation, and I have a mild preference that we stay consistent with that:

https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Comments#Comments

**Comment by cgordon** [comment]
Apologies! I lost track of this change, merging now..

**Pull request merged**